### DISCUSSION TITLE
Policies - users with non-matching user_id can still see everything

### DISCUSSION THREAD

**Discussion by AlexanderCryptoOwl** [supabase/supabase#5476] [Questions] [answered]
Hi, 

I have this policy code

```
(EXISTS ( SELECT profiles.created_at,
    profiles.is_subscribed
   FROM profiles
  WHERE ((profiles.is_subscribed = true) OR ((profiles.created_at > (CURRENT_DATE - '7 days'::interval)) AND (uid() = notifications.user_id)))))
```

It looks like the OR logic works based on my testing but for some reason users that matches the OR statement is able to see all notifications even the ones not associated with their user_id?

**Comment by GaryAustin1** [comment, marked as answer]
I believe (by counting parenthesis) your AND is only applied to the 2nd 1/2 of the OR (profiles.create_at >...  Not sure that is what you intend as then you will return any row with is_subscribe=true for any user.

**Reply by AlexanderCryptoOwl** [reply]
Thanks, for it was a late last night adjusted it to this:

(EXISTS ( SELECT profiles.created_at, profiles.is_subscribed FROM profiles WHERE (((uid() = notifications.user_id) AND (profiles.is_subscribed = true)) OR (profiles.created_at > (CURRENT_DATE - '7 days'::interval)))))
